src/iterators.rs

```diff
@@ -641,3 +640,157 @@ pub fn new_axis_iter_mut<A, D>(v: ArrayViewMut<A, D>,
         life: PhantomData,
     }
 }
+
+/// An iterator that traverses over the specified axis
+/// and yields views of the specified size on this axis.
+///
+/// For example, in a 2 × 8 × 3 array, if the axis of iteration
+/// is 1 and the chunk size is 2, the yielded elements
+/// are 2 × 2 × 3 views (and there are 4 in total).
+///
+/// Iterator element type is `ArrayView<'a, A, D>`.
+pub struct ChunkIter<'a, A: 'a, D> {
+    iter: OuterIterCore<A, D>,
+    last_ptr: *mut A,
+    last_dim: D,
+    life: PhantomData<&'a A>,
+}
+
+fn chunk_iter_parts<A, D: Dimension>(v: ArrayView<A, D>, axis: usize, size: usize)
+    -> (OuterIterCore<A, D>, *mut A, D)
+{
+    let axis_len = v.shape()[axis];
+    let size = if size > axis_len { axis_len } else { size };
+    let last_index = axis_len / size;
+    let rem = axis_len % size;
+    let shape = if rem == 0 { last_index } else { last_index + 1 };
+    let stride = v.strides()[axis] * size as isize;
+
+    let mut inner_dim = v.dim.clone();
+    inner_dim.slice_mut()[axis] = size;
+
+    let mut last_dim = v.dim.clone();
+    last_dim.slice_mut()[axis] = if rem == 0 { size } else { rem };
+
+    let last_ptr = if last_index < axis_len {
+        unsafe {
+            v.ptr.offset(stride * last_index as isize)
```

Comment:
Actually this won't work either, I think there's a possibility to index out of bounds if `rem == 0`. That's because any time the size divides the dimension length, `last_index` is not a "valid" index (ie `last_index * stride >= axis_len`).

So it seems the proper test is `rem != 0`.